### DISCUSSION TITLE
NumberInput: Formatter and Parser not available in v7

### DISCUSSION THREAD

**Discussion by jsjohann** [mantinedev/mantine#4938] [Q&A]
### What package has an issue

@mantine/core

### Describe the bug

In Mantine v6, it was possible to use the formatter/parser prop of the NumberInput component to manipulate the displayed value of the NumberInput. That feature seems to be not available anymore in Mantine v7 or did I miss anything? (https://mantine.dev/core/number-input/?t=props). Will it be back in a future release or is there any workaround available? 

Thank you!

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

7.0.2

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
You can find 7.0.0 changelog here – https://mantine.dev/changelog/7-0-0/
NumberInput was migrated to `react-number-format`. You should be able to achieve everything that you've done with parser and formatter with other props. Props from previous versions are not planned to be added back.

**Reply by jsjohann** [reply]
Thanks for the hint! But the props of react-number-format doesn't seem to be available in a Typescript context? I mean, the docs of Mantine do not say anything about customization using react-number-format props.

Edit: I did just realize that there are props from react-number-format. But I can't get a custom format to work?

**Reply by rtivital** [reply]
What format do you need?

**Reply by jsjohann** [reply]
Basically, I would like to enter a frame number of a video and would like to display it in the form hh:mm:ss:frame (e.g. frame 30 of a 25fps video would be shown as 00:00:01:5).

**Reply by rtivital** [reply]
It is better to use mask input with InputBase instead of NumberInput – https://mantine.dev/core/input/#inputbase-component

**Reply by jsjohann** [reply]
I had a look into it and got a semi-usable implementation. The problem is, that I like to use the increment/decrement buttons of the base NumberInput. Using `InputBase` with `component={NumberFormatBase}` to access the react-number-format props `format` and `removeFormatting` (should be the same as the old v6 NumberInput props `formatter` and `parser`), doesn't support me on having the standard props for editing numbers like `min`, `max`, `step` or the clamp handling. Basically, I was able in Mantine v6 to click on the increment/decrement handlers of the NumberInput to adjust the frame-value by one and the current time gets formatted accordingly. Additionally to that, I can directly change the minutes value of the formatted time string to edit the underlying frame number. Also, editing the formatted time string doesn't work as smooth like it did in the previous native Mantine implementation.

**Reply by jsjohann** [reply]
I ended up writing a custom component based on the InputBase-component which mimics the functionality/handling as well as the look of the NumberInput (increment/decrement handlers, clamping of values between min/max). It uses an internal value state for the formatted number (frame number to hh:mm:ss::ff presentation) and vice versa the parsed value for further handling when the value is changed. 

Would be awesome if it is possible to achieve the same with the NumberInput (e.g. providing a custom increment/decrement function to the existing handlers as well as complete control over the displayed value).
